### ISSUE TITLE
[ladeco] Convert LaDeCo to a Hugging Face processor and inference recipe

### ISSUE THREAD

**Issue by shunk031** [creative-graphic-design/design-generators#24]
Reuse LaDeCo's released `microsoft/elem2design` Hugging Face model and implement the layered graphic-design processor, schema validation, and inference recipe needed to reproduce the paper's layer-by-layer design composition workflow with standard HF loading.

## Approach

- Use the public HF model repo as the weight source: README links https://huggingface.co/microsoft/elem2design and says the released model is trained on the public Crello dataset; it also notes a stronger private-data model is not released.
- The original command-line inference described in README is `CUDA_VISIBLE_DEVICES=0 python llava/infer/infer.py --model_name_or_path /path/to/model/checkpoint-xxxx --data_path /path/to/data/test.json --image_folder /path/to/crello_images --output_dir /path/to/output_dir --start_layer_index 0 --end_layer_index 4`, and the demo is `python app/app.py --model_name_or_path /path/to/model/checkpoint-xxxx`.
- The concrete loading helper in `thirdparty/opencole/opencole/model/llava.py::load_llava()` already uses HF classes: `LlavaForConditionalGeneration.from_pretrained(pretrained_model_name_or_path, device_map="auto", torch_dtype=torch.float16, low_cpu_mem_usage=True, use_safetensors=True, quantization_config=BitsAndBytesConfig(...))` plus `LlavaProcessor.from_pretrained(pretrained_model_name_or_path)`.
- The inference launcher `thirdparty/opencole/opencole/inference_launcher/typographylmm.py` builds a `LayoutManager`, a LangChain `PydanticOutputParser` over `layoutlib.schema.get_layout_pydantic_model(...)`, and a `TypographyLMMTester`; it reads `crello_default.json`, pairs detail JSON files with rendered PNGs, and writes one output JSON per design.
- Preserve LaDeCo's layer-planning contract in the processor: README defines the layer indices as `0: Background`, `1: Underlay`, `2: Logo/Image`, `3: Text`, `4: Embellishment`, and training data has five dialogue rounds that progressively predict attributes from background to embellishment.
- Training/config are LLaVA/LoRA based: `scripts/finetune_lora.sh` trains with `--lora_enable True`, `--model_name_or_path $BASE_MODEL`, `--vision_tower openai/clip-vit-large-patch14-336`, `--mm_projector_type mlp2x_gelu`, `--image_aspect_ratio pad`, `--model_max_length 15000`, and LoRA rank/alpha arguments; README example uses `meta-llama/Llama-3.1-8B`.
- Determinism support exists for preprocessing but must be surfaced for generation: `opencole.util.set_seed()` seeds Python, NumPy, and Torch, while inference uses LLaVA generation through `TypographyLMMTester`; the recipe should expose seed and generation kwargs and record any sampling defaults used by the tester.
- Keep heavy pins isolated: `pyproject.toml` requires `torch==2.7.0`, `torchvision==0.22.0`, `transformers==4.44.2`, `tokenizers==0.19.1`, `accelerate==0.34.2`, `numpy==1.26.4`, `scikit-learn==1.5.1`, `datasets==3.6.0`, `gradio`, and optional training dependencies `deepspeed==0.14.4`, `ninja`, and `wandb==0.18.1`; `flash-attn` is installed separately for training.
- Recommended API matches issue #2: reuse HF LLaVA classes and provide a LaDeCo-specific processor/schema layer rather than converting model internals.

## Steps

- [ ] Create a `models/ladeco/` workspace member with a processor/schema package and an optional `vendor` extra for OpenCOLE/LLaVA training reproduction.
- [ ] Add a `microsoft/elem2design` loading recipe using `LlavaForConditionalGeneration` and `LlavaProcessor`, including optional 4-bit/8-bit quantization flags.
- [ ] Implement the layered design input/output processor for background, underlay, logo/image, text, and embellishment rounds, reusing Pydantic layout schemas for structured outputs.
- [ ] Add tests for layer-index mapping, schema validation, padding layout transforms, and parser failure handling without weights.
- [ ] Add optional parity tests that invoke the original inference launcher on one detail/image pair and compare normalized layout JSON.

## References

- Paper: https://arxiv.org/abs/2412.19712
- Weights: https://huggingface.co/microsoft/elem2design
- Original implementation: vendor/ladeco (MIT)

**Comment by shunk031** [comment]
Data note: Crello is not yet available in the `creative-graphic-design` HF org (verified 2026-07-17, see #2). For now, use the dataset exactly as the original repository distributes it (vendor README / download scripts).

- [ ] TODO: switch the data-loading path to a HF Hub download once Crello is imported into the org. Keep all dataset loading behind this model's processor so the swap stays localized (interim rule in #2's Data source policy).

**Comment by shunk031** [comment]
Data note update: for Crello, use the official HF dataset [`cyberagent/crello`](https://huggingface.co/datasets/cyberagent/crello) as the canonical source (owner decision; supersedes the earlier "original distribution until org import" note for Crello). Keep loading behind the processor; vendor/starter-kit Crello assets remain valid for exact-parity fixtures where the authors' processed splits matter.
